### DISCUSSION TITLE
Need help sampling the flux space of metabolic networks

### DISCUSSION THREAD

**Discussion by Bastien-N** [GeomScale/dingo#42] [Q&A] [answered]
Hi all,

I am trying to use volesti to sample the flux space of metabolic networks but am having some difficulties. I was using the example script and the volesti version from the branch SoCG21 (indicated in the paper "Geometric algorithms for sampling the flux space of metabolic networks"). However, this failed to sample the model in over 10H, even though it is a fair bit smaller than the Recon2D model which could be sampled in 4H according to the paper.

I then tried to use the current volesti version directly, but the tutorials available are quite sparse and I do not understand how to construct a polytope from a model (Hpolytope$new() does not appear to work in this version), or how to perform any of the preprocessing steps.

Would anyone have a simple explanation for how to sample the flux space of a large metabolic network using the current version of volesti?

**Comment by vissarion** [comment, marked as answer]
Hi @Bastien-N, 

most of the functionalities about metabolic models are moved to the new package [dingo](https://github.com/GeomScale/dingo). There is a [tutorial](https://github.com/GeomScale/dingo/blob/develop/dingo_tutorial.ipynb) that describes how to sample the flux space of a metabolic network. Please let us know if this is OK for you or you need more help.

**Reply by Bastien-N** [reply]
Thank you for your answer. I was hoping to be able to avoid using other lanquages than R and Matlab as these are those me and my collaborators are already using in this project.

I will try dingo regardless and come back to you with any issue.

**Comment by vissarion** [comment]
OK then you could open a feature request in volesti (as an issue) describing your use-case and needs. But I think the most quick and efficient way right now is to try dingo.
